{
  final Session session=openSession();
  session.beginTransaction();
  session.doWork(new Work(){
    @Override public void execute(    Connection connection) throws SQLException {
      PreparedStatement ps=null;
      try {
        ps=((SessionImplementor)session).getJdbcCoordinator().getStatementPreparer().prepareStatement("INSERT INTO T_MEMBERSHIP (user_id, group_id) VALUES (?, ?)");
        ps.setLong(1,52134241);
        ps.setLong(2,5342);
        ((SessionImplementor)session).getJdbcCoordinator().getResultSetReturn().executeUpdate(ps);
        fail("INSERT should have failed");
      }
 catch (      ConstraintViolationException ignore) {
      }
 finally {
        releaseStatement(session,ps);
      }
    }
  }
);
  session.getTransaction().rollback();
  session.close();
}
